src/core/exec-invoke.c

```diff
@@ -1287,7 +1289,7 @@ static int setup_pam(
 
         parent_pid = getpid_cached();
 
-        r = safe_fork("(sd-pam)", 0, NULL);
+        r = pidref_safe_fork("(sd-pam)", 0, &child_pidref);
```

Comment:
um, given we still lack the kernel API to attach processes to cg via pidfd, kinda tempted to make pam subprocess attach itself rather than in the parent.

We should really sort #31033 out (maybe add a comment about it). But until then, it's probably the only way to ensure this is race-free